evalml/data_checks/data_check_action.py

```diff
@@ -0,0 +1,24 @@
+class DataCheckAction:
+    """Base class for all DataCheckActions."""
+
+    def __init__(self, action_code, details=None):
+        """
+        A recommended action returned by a DataCheck.
+
+        Arguments:
+            action_code (DataCheckActionCode): Action code associated with the action.
```

Comment:
Since the action recommendation process will be interactive, I think it will be easier for users to modify the actions if they can specify the action code via string rather than having to import the `DataCheckActionCode`, similar to what we do for problem types. 

```python
data_check = DefaultDataChecks()
actions = data_check.validate(X, y)["actions"]
actions[0].action_code = "do_nothing"
```

as opposed to

```python
from evalml.datachecks import DataCheckActionCode
data_check = DefaultDataChecks()
actions = data_check.validate(X, y)["actions"]
actions[0].action_code = DataCheckActionCode.DO_NOTHING
```

Let me know what you think. This is out of scope for this PR but I think we should make the actions easy to modify as we build them.

Reply:
Filed #1904 :)